packages/fluentui/react-northstar/src/components/RadioGroup/RadioGroupItem.tsx

```diff
@@ -76,93 +84,124 @@ export const radioGroupItemSlotClassNames: RadioGroupItemSlotClassNames = {
   indicator: `${radioGroupItemClassName}__indicator`,
 };
 
-class RadioGroupItem extends AutoControlledComponent<WithAsProp<RadioGroupItemProps>, RadioGroupItemState> {
-  elementRef = React.createRef<HTMLElement>();
-
-  static create: ShorthandFactory<RadioGroupItemProps>;
-
-  static displayName = 'RadioGroupItem';
-
-  static deprecated_className = radioGroupItemClassName;
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false,
-    }),
-    checked: PropTypes.bool,
-    defaultChecked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    indicator: customPropTypes.shorthandAllowingChildren,
-    label: customPropTypes.itemShorthand,
-    name: PropTypes.string,
-    onClick: PropTypes.func,
-    onChange: PropTypes.func,
-    shouldFocus: PropTypes.bool,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    vertical: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    accessibility: radioGroupItemBehavior,
-    indicator: {},
+export type RadioGroupItemStylesProps = Required<Pick<RadioGroupItemProps, 'disabled' | 'vertical' | 'checked'>>;
+
+const RadioGroupItem: React.FC<WithAsProp<RadioGroupItemProps>> &
+  FluentComponentStaticProps<RadioGroupItemProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(RadioGroupItem.displayName, context.telemetry);
+  setStart();
+  const { label, indicator, disabled, vertical, className, design, styles, variables } = props;
+  const elementRef = React.useRef<HTMLElement>();
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(RadioGroupItem.handledProps, props);
+
+  const [checked, setChecked] = useAutoControlled({
+    defaultValue: props.defaultChecked,
+    value: props.checked,
+    initialValue: false,
+  });
+
+  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+    _.invoke(props, 'onClick', e, props);
+    setChecked(prevChecked => !prevChecked);
   };
 
-  static autoControlledProps = ['checked'];
-
-  actionHandlers = {
-    performClick: e => {
-      e.preventDefault();
-      this.handleClick(e);
+  React.useEffect(() => {
+    checked && props.shouldFocus && elementRef.current.focus();
```

Comment:
```suggestion
    if(checked && shouldFocus) elementRef.current.focus();
```

`shouldFocus` is also a dependency for the hook

Reply:
done